Sources/Load.php

```diff
@@ -3410,60 +3413,84 @@ function loadDatabase()
 /**
  * Try to load up a supported caching method. This is saved in $cacheAPI if we are not overriding it.
  *
- * @param string $overrideCache Try to use a different cache method other than that defined in $cache_accelerator.
+ * @param array $overrideCache Try to use a different cache method other than that defined in $cache_accelerator.
  * @param bool $fallbackSMF Use the default SMF method if the accelerator fails.
  * @return object|false A object of $cacheAPI, or False on failure.
  */
-function loadCacheAccelerator($overrideCache = null, $fallbackSMF = true)
+function loadCacheAccelerator($overrideCache = [], $fallbackSMF = true)
 {
-	global $sourcedir, $cacheAPI, $cache_accelerator, $cache_enable;
+	global $cacheAPI, $cache_accelerator, $cache_enable, $cacheAPIdir;
 
-	// is caching enabled?
+	// Is caching enabled?
 	if (empty($cache_enable) && empty($overrideCache))
 		return false;
 
 	// Not overriding this and we have a cacheAPI, send it back.
 	if (empty($overrideCache) && is_object($cacheAPI))
 		return $cacheAPI;
+
 	elseif (is_null($cacheAPI))
 		$cacheAPI = false;
 
-	// Make sure our class is in session.
-	require_once($sourcedir . '/Class-CacheAPI.php');
+	// Autoload hasn't been called yet :/
+	require_once($cacheAPIdir .'/CacheApi.php');
+	require_once($cacheAPIdir .'/CacheApiInterface.php');
+
+	$apis_dir = $cacheAPIdir .'/'. CacheApi::APIS_FOLDER;
 
 	// What accelerator we are going to try.
-	$tryAccelerator = !empty($overrideCache) ? $overrideCache : (!empty($cache_accelerator) ? $cache_accelerator : 'smf');
-	$tryAccelerator = strtolower($tryAccelerator);
+	$cache_file_info = !empty($overrideCache) ? $overrideCache : (!empty($cache_accelerator) ?
+		smf_json_decode($cache_accelerator, true) : array(
+			'class_name' => CacheApi::APIS_DEFAULT,
+			'file_info' => array(
+				'basename' => CacheApi::APIS_DEFAULT .'.php',
+			),
+			'txt_key' => strtolower(CacheApi::APIS_DEFAULT),
+		));
+
+	$file_to_load = $apis_dir . '/' . $cache_file_info['file_info']['basename'];
```

Comment:
Not a fan of storing arrays in Settings.php. IMO short strings work the best  as  they are quick and easy for intermediate/advanced forum owner to change. (Also devs)

Reply:
But that's one of the key features of json strings, they are somehow easier to read than a serialized one.

Reply:
I see you quickly put it  back so I'm not going to continue this argument. 😃

Reply:
mmm not sure what you mean, I didn't modify this part of the code on recent commits. It still stores a json encoded array.

Reply:
oh... when you pushed d80dacbcb1569fea41cea1ae03aa6d336dd8491e, several commits got lost.

Reply:
Yeah I use rebase instead of merging, thats why old commits appears to be new ones, in a sense they are new (to the release-2.1 branch) but originally they are old.

Reply:
Oh my!  you are right a lot of commits got lost 😱 will check it out why.